src/main/java/uk/gov/justice/hmpps/datacompliance/services/client/image/recognition/AwsImageRecognitionClient.java

```diff
@@ -0,0 +1,95 @@
+package uk.gov.justice.hmpps.datacompliance.services.client.image.recognition;
+
+import com.amazonaws.services.rekognition.AmazonRekognition;
+import com.amazonaws.services.rekognition.model.*;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Optional;
+
+import static com.amazonaws.services.rekognition.model.QualityFilter.HIGH;
+import static java.lang.String.format;
+
+@Slf4j
+@Component
+@ConditionalOnProperty(name = "image.recognition.provider")
+public class AwsImageRecognitionClient implements ImageRecognitionClient {
+
+    private final String collectionId;
+    private final AmazonRekognition client;
+
+    public AwsImageRecognitionClient(final AmazonRekognition client,
+                                     @Value("${image.recognition.aws.collection.id}") final String collectionId) {
+
+        log.info("Configured to use AWS Rekognition for image recognition");
+
+        this.client = client;
+        this.collectionId = collectionId;
+    }
+
+    @Override
+    public Optional<String> uploadImageToCollection(final byte[] imageData,
+                                                    final String offenderNumber,
+                                                    final long imageId) {
+
+        log.debug("Uploading image data for offender: '{}', image: '{}'", offenderNumber, imageId);
+
+        final var indexedFaces = client.indexFaces(
+                generateIndexFaceRequest(imageData, offenderNumber, imageId))
+                .getFaceRecords();
+
+        return ensureOnlyOneFaceIndexed(offenderNumber, imageId, indexedFaces);
+    }
+
+    private Optional<String> ensureOnlyOneFaceIndexed(final String offenderNumber,
+                                                      final long imageId,
+                                                      final List<FaceRecord> indexedFaces) {
+
+        if (indexedFaces.size() != 1) {
+            log.warn("Face count: '{}' for offender: '{}' and image: '{}'", indexedFaces.size(), offenderNumber, imageId);
```

Comment:
took me a while to realise you were deleting all the faces as you wouldn't be able to tell which face was the offender face - might be worth adding a comment or explaining in the warning log.